packages/webview_flutter/example/test_driver/webview.dart

```diff
@@ -54,6 +54,46 @@ void main() {
     final String currentUrl = await controller.currentUrl();
     expect(currentUrl, 'https://www.google.com/');
   });
+
+  group('error handling', () {
+    test('invalid host', () async {
+      final Completer<WebViewError> errorCompleter = Completer<WebViewError>();
+      testWebViewError('https://example.invalid/', errorCompleter);
+      final WebViewError error = await errorCompleter.future;
+      expect(error, isNotNull);
+      expect(error.isConnectError, true);
+      expect(error.connectErrorType, WebViewConnectErrorType.connect);
+    });
+    test('internal server error', () async {
+      final Completer<WebViewError> errorCompleter = Completer<WebViewError>();
+      testWebViewError('https://httpstat.us/500', errorCompleter);
+      final WebViewError error = await errorCompleter.future;
+      expect(error, isNotNull);
+      expect(error.isConnectError, false);
+      expect(error.statusCode, 500);
+    });
+  });
+}
+
+Future<void> testWebViewError(
```

Comment:
From talking to Amir I get the sense that he prefers not to have helper methods in tests, and instead to have really verbose tests with a lot of code duplication.

Reply:
okay, i have inlined the helper method: https://github.com/flutter/plugins/pull/1389/commits/05444faa184d53c1a1d008488d17f7aa3498f5b2
if you figure it's easier to figure out whats going on here, it's fine with me.. my personal taste would be to have the helper method, because this seems so verbose that it's actually harder to grasp whats going on (although "whats going on" for me, is knowing where the differences are between inputs vs. outputs, with so much boiler plate it's actually hard to see the different cases being tested..)
if it's okay this way, okay.. otherwise i can also revert it and change it to `Future<WebViewError>` whatever fits better..